virttest/qemu_devices/qcontainer.py

```diff
@@ -942,6 +942,27 @@ def get_aavmf_vars(params):
                                                   parent_bus={'aobject': 'pci.0'}))
             return devices
 
+        def machine_s390x_virtio(cmd=False):
+            """
+            s390x (s390) doesn't support PCI bus.
+            :param cmd: If set uses "-M $cmd" to force this machine type
+            :return: List of added devices (including default buses)
+            """
+            devices = []
+            # Add virtio-bus
+            # TODO: Currently this uses QNoAddrCustomBus and does not
+            # set the device's properties. This means that the qemu qtree
+            # and autotest's representations are completelly different and
+            # can't be used.
+            logging.warn('Support for s390x is highly experimental!')
+            bus = qbuses.QNoAddrCustomBus('bus', [['addr'], [64]],
```

Comment:
100% improvement in number of slots since last PR :smile: I tried the upstream version and got up to 8000 devices, do you know the actual limit? (no need to change anything, the limit 64 is fine for now, I'm only curious)